internal/controller/imageupdater_controller.go

```diff
@@ -74,6 +75,11 @@ type ImageUpdaterReconciler struct {
 	Wg sync.WaitGroup
 }
 
+const (
+	// imageUpdaterFinalizer is the name of the finalizer used by the ImageUpdater controller.
+	imageUpdaterFinalizer = "argocd-image-updater.argoproj.io/finalizer"
```

Comment:
Shall we follow argo-cd finalizers' naming, like https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/application_defaults.go#L10 ? It makes it easier to have more granular finalizers if needed.